lighthouse-core/audits/consistently-interactive.js

```diff
@@ -0,0 +1,247 @@
+const SCORING_MEDIAN = 10000;
+const SCORING_TARGET = 5000;
+
+const REQUIRED_QUIET_WINDOW = 5000;
+const ALLOWED_CONCURRENT_REQUESTS = 2;
+
+const distribution = TracingProcessor.getLogNormalDistribution(
+  SCORING_MEDIAN,
+  SCORING_POINT_OF_DIMINISHING_RETURNS
+);
+
+/**
+ * @fileoverview This audit identifies the time the page is "consistently interactive".
+ * Looks for the first period of at least 5 seconds after FMP where both CPU and network were quiet,
+ * and returns the timestamp of the beginning of the CPU quiet period.
+ * @see https://docs.google.com/document/d/1GGiI9-7KeY3TPqS3YT271upUVimo-XiL5mwWorDUD4c/edit#
+ */
+class ConsistentlyInteractiveMetric extends Audit {
+  /**
+   * @return {!AuditMeta}
+   */
+  static get meta() {
+    return {
+      category: 'Performance',
+      name: 'consistently-interactive',
+      description: 'Consistently Interactive (beta)',
+      helpText: 'The point at which most network resources have finished loading and the ' +
+          'CPU is idle for a prolonged period.',
+      optimalValue: SCORING_TARGET.toLocaleString() + 'ms',
+      scoringMode: Audit.SCORING_MODES.NUMERIC,
+      requiredArtifacts: ['traces', 'networkRecords']
+    };
+  }
+
+  /**
+   * @param {!Array} networkRecords
+   * @param {{timestamps: {traceEnd: number}}} traceOfTab
+   * @return {!Array<{start: number, end: number}>}
+   */
+  static _findNetworkQuietPeriods(networkRecords, traceOfTab) {
+    const traceEnd = traceOfTab.timestamps.traceEnd;
+
+    // First collect the timestamps of when requests start and end
+    const timeBoundaries = [];
+    networkRecords.forEach(record => {
+      const scheme = record.parsedURL && record.parsedURL.scheme;
+      if (scheme === 'data' || scheme === 'ws') {
+        return;
+      }
+
+      // convert the network record timestamp to ms to line-up with traceOfTab
+      timeBoundaries.push({time: record.startTime * 1000, isStart: true});
+      timeBoundaries.push({time: record.endTime * 1000, isStart: false});
+    });
+
+    timeBoundaries.sort((a, b) => a.time - b.time);
+
+    let numInflightRequests = 0;
+    let quietPeriodStart = 0;
+    const quietPeriods = [];
+    timeBoundaries.forEach(boundary => {
+      if (boundary.isStart) {
+        // we've just started a new request. are we exiting a quiet period?
+        if (numInflightRequests === ALLOWED_CONCURRENT_REQUESTS) {
+          quietPeriods.push({start: quietPeriodStart, end: boundary.time});
+          quietPeriodStart = Infinity;
+        }
+        numInflightRequests++;
+      } else {
+        numInflightRequests--;
+        // we've just completed a request. are we entering a quiet period?
+        if (numInflightRequests <= ALLOWED_CONCURRENT_REQUESTS) {
+          quietPeriodStart = Math.min(boundary.time, quietPeriodStart);
+        }
+      }
+    });
+
+    // Check if the trace ended in a quiet period
+    if (quietPeriodStart !== Infinity) {
```

Comment:
> just fixed, adding the request on start instead and checking `.finished`

this is the right solution, but nothing else that depends on network records is checking for this as they all could assume that all requests were finished when they were written. They're often just taking all of them and filtering on whatever they need to look for. We need to do a survey of all the uses, fix where it makes sense, add unit tests like you added in a5d62b8 for CI, etc